app/main.py

```diff
@@ -1,3 +1,64 @@
+from __future__ import annotations
+
+from typing import Union, Callable
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, int, float]) -> Distance:
+        return self._math_operation(other, lambda a, b: a + b)
+
+    def __iadd__(self, other: Union[Distance, int, float]) -> Distance:
+        return self._math_operation(other, lambda a, b: a + b)
+
+    def __mul__(self, other: Union[int, float]) -> Distance:
+        if isinstance(other, (int, float)):
+            return self._math_operation(other, lambda a, b: a * b)
+
+    def __truediv__(self, other: Union[Distance, int, float]) -> Distance:
+        if isinstance(other, (int, float)):
+            return self._math_operation(
+                other, lambda a, b: a / b, rounded=True
+            )
```

Comment:
Must create new instance not change self
Also for this function no need to reuse math_operations just write:
return Distance(other * self.km)